sbom-utility-scripts/scripts/add-image-reference-script/add_image_reference.py

```diff
@@ -139,16 +149,25 @@ def update_component_in_cyclonedx_sbom(sbom: dict, image: Image) -> dict:
     Returns:
         dict: Updated SBOM with the image reference added.
     """
-    # Add the image component to the components list
     image_component = {
         "type": "container",
         "name": image.name,
         "purl": image.purl(),
         "version": image.tag,
         "hashes": [{"alg": image.digest_algo_cyclonedx, "content": image.digest_hex_val}],
     }
-    sbom["components"].insert(0, image_component)
-    sbom["metadata"]["component"] = image_component
+    if image.build_image:
+        # Add base image property to image_component
+        image_component["properties"]: [{
+            "name": "konflux:container:is_builder_image:for_stage",
+            "value": "1",
```

Comment:
I think hardcoding this as 1 wouldn't be correct, since this can be added to an image that already has several `konflux:container:is_builder_image:for_stage`, and potentially have one that is already stage 1. Maybe we'll need to increment this accordingly?

Reply:
Yeah. A hardcoded 1 is wrong. the discussion about the property will condition what we do here :)
https://github.com/konflux-ci/build-tasks-dockerfiles/pull/254#issuecomment-2717609451